### DISCUSSION TITLE
Ways to know which field in useFieldArray triggered change

### DISCUSSION THREAD

**Discussion by ap-justin** [react-hook-form/react-hook-form#8963] [Q&A]
when using `useFieldArray` is there a way to know which `field` triggered a change? 
```ts
const { fields, append, remove } = useFieldArray<SomeType>({
    name: "fields", // unique name for your Field Array
  })
  
 fields.map(field => <input />) 
//e.g
input 1 
input 2
input 3 //know that I typed here
  
```
these `n` inputs should sum up to `100`, say `a50, b20, c20`, if I make, `c` more than `30`, the sum would be `>100` and would want to show that field `c` is the cause of the overflow

**Comment by bluebill1049** [comment]
if you want to know which input is getting onChange, perhaps you should attach custom event on the `onChange` prop, then you know which input is triggering the event.

**Reply by ap-justin** [reply]
thanks for the idea ser @bluebill1049
